### PR TITLE
[Merged by Bors] - perf(Topology.Algebra.Module): factor out `smul`

### PR DESCRIPTION
**Pull request by mattrobball**
We want to avoid making Lean unfold `smul` during unification. A separate instance does helps at the cost of some elaboration failures.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by mattrobball** [comment]
!bench

File: Mathlib/Analysis/SpecialFunctions/Trigonometric/Deriv.lean [hunk 1]
```diff
@@ -338,7 +338,8 @@ theorem fderivWithin_ccos (hf : DifferentiableWithinAt ℂ f s x) (hxs : UniqueD
   hf.hasFDerivWithinAt.ccos.fderivWithin hxs
 #align fderiv_within_ccos fderivWithin_ccos
 
-@[simp]
+@[simp, nolint simpNF] -- `simp` times out trying to find `Module ℂ (E →L[ℂ] ℂ)` but this fine
+-- with all of `Mathlib` opened -- no idea why
```
Comment (YaelDillies) [inline Mathlib/Analysis/SpecialFunctions/Trigonometric/Deriv.lean:342]: missing a word?

Comment (mattrobball) [reply Mathlib/Analysis/SpecialFunctions/Trigonometric/Deriv.lean]: Better english now hopefully

**Review by YaelDillies — APPROVED** [review]
That looks fine but what's the interaction with https://leanprover.zulipchat.com/#narrow/stream/144837-PR-reviews/topic/.2311521.20speeding.20up.20lots.20of.20instances ? Won't that discussion make this PR obsolete soon?

**Comment by mattrobball** [comment]
`fast_instsnces%` doesn't chunk up it's input into new declarations and add them to the environment. It only looks for preexisting instances. So if you don't split off, `SMul` here there won't be any preexisting instances to find. 

So it in fact works in tandem.

**Comment by YaelDillies** [comment]
Okay, let's go

maintainer merge

**Review by jcommelin — APPROVED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**